.github/workflows/main.yml

```diff
@@ -43,6 +43,9 @@ jobs:
           fetch-depth: 0
 
       # Install dependencies
+      - name: Remove pre-existing dotnet sdk 8 installs
+        run: sudo apt remove --purge dotnet-sdk-8.0
+        working-directory: packages/blazor-workspace/
       - uses: actions/setup-dotnet@v4
```

Comment:
>If the image version of the sdk is newer, then even if we install a locked older version of the SDK, our lockfile checks will fail

Just to double check, what mechanism did you use to install a locked older version? Was it [specifying an exact version using the `dotnet-version` parameter](https://github.com/actions/setup-dotnet?tab=readme-ov-file#supported-version-syntax)? If you haven't tried that already then it might be worth a go.

Reply:
We are using the [`global-json-file`](https://github.com/actions/setup-dotnet?tab=readme-ov-file#using-the-global-json-file-input) option.
I'm skeptical that using `dotnet-version` would behave differently (I don't know why it would try and uninstall old versions). It would also lead to duplicating the version used in two spots which is a different trade-off I'm not a huge fan of. If interested in that approach I can investigate it in a follow-on.